expression/builtin_other.go

```diff
@@ -1027,8 +1046,10 @@ func (b *builtinGetIntVarSig) evalInt(row chunk.Row) (int64, bool, error) {
 	varName = strings.ToLower(varName)
 	sessionVars.UsersLock.RLock()
 	defer sessionVars.UsersLock.RUnlock()
-	if v, ok := sessionVars.Users[varName]; ok {
-		return v.GetInt64(), false, nil
+	if useVar, ok := sessionVars.UserVars.Vars[varName]; ok {
+		if v, ok1 := useVar.(*Constant); ok1 {
+			return v.Value.GetInt64(), false, nil
```

Comment:
We can use `Constant.EvalInt` directly here, `return var.(*Constant).EvalInt(ctx, chunk.Row{})`